### PR TITLE
add write-good support

### PR DESCRIPTION
**Pull request by jondot**
Here: http://github.com/jondot/write-good (my own fork, until upstream accepts the parse additions that make it work with neomake)

### CODE REVIEW

File: autoload/neomake/makers/ft/markdown.vim [hunk 1]
```diff
@@ -19,6 +19,13 @@ function! neomake#makers#ft#markdown#proselint()
                 \ }
 endfunction
 
+function! neomake#makers#ft#markdown#writegood()
+    return {
+                \ 'args': ['--parse'],
+                \ 'errorformat': '%f:%l:%c:%m'
```
Comment (blueyed) [inline autoload/neomake/makers/ft/markdown.vim]: Should specify the type, e.g. %W.
See also https://github.com/neomake/neomake/issues/584.

File: autoload/neomake/makers/ft/text.vim [hunk 2]
```diff
@@ -1,5 +1,12 @@
 function! neomake#makers#ft#text#EnabledMakers()
-    return ['proselint']
+    return ['proselint', 'writegood']
+endfunction
+
+function! neomake#makers#ft#text#writegood()
+    return {
+                \ 'args': ['--parse'],
+                \ 'errorformat': '%f:%l:%c:%m'
```
Comment (blueyed) [inline autoload/neomake/makers/ft/text.vim]: Also type is missing here.

File: autoload/neomake/makers/ft/markdown.vim (hunk 1 above)
Comment (jondot) [inline autoload/neomake/makers/ft/markdown.vim]: Thanks for letting me know - is it possible for you to point me to documentation? I'd like to know how %W works. So far I've used other plugins as guidance.

Comment (blueyed) [inline autoload/neomake/makers/ft/markdown.vim]: Check `:h errorformat`.

Comment (jondot) [inline autoload/neomake/makers/ft/markdown.vim]: Got it. so the message is not multiline. does this still apply?

Comment (blueyed) [inline autoload/neomake/makers/ft/markdown.vim]: Yes.

**Comment by jondot** [comment]
should be good now

File: autoload/neomake/makers/ft/text.vim [hunk 3]
```diff
@@ -1,5 +1,12 @@
 function! neomake#makers#ft#text#EnabledMakers()
-    return ['proselint']
+    return ['proselint', 'writegood']
+endfunction
+
+function! neomake#makers#ft#text#writegood()
+    return {
```
Comment (blueyed) [inline autoload/neomake/makers/ft/text.vim]: This should use/return `neomake#makers#ft#markdown#writegood()`.
Then it's only defined in one place.

**Comment by blueyed** [comment]
Please also add it to the wiki: https://github.com/neomake/neomake/wiki/Makers, and report back.
I'll merge it then.

**Comment by blueyed** [comment]
Thanks!
Closing in favor of https://github.com/neomake/neomake/pull/760#issuecomment-260165097 / the new PR to come.

**Comment by blueyed** [comment]
Now we're at https://github.com/neomake/neomake/pull/794.

**Pull request closed without merging**